### ISSUE TITLE
[Bug] examples/example-stateful-function-config.yaml does not exist in 2.11 distro

### ISSUE THREAD

**Issue by anguillanneuf** [apache/pulsar#20063]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

The Get Started guide for [stateful functions](https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-stateful-functions) can't be followed because this file `examples/example-stateful-function-config.yaml` does not appear to exist. 

![Screenshot 2023-04-10 at 4 33 25 PM](https://user-images.githubusercontent.com/9518298/231018710-6b277b80-c352-46bd-9871-cc7cfe5de2ed.png)

```sh
tz@pantone:~/apache-pulsar-2.11.0$ ls examples/
api-examples.jar  example-function-config.yaml  example-window-function-config.yaml  python-examples
```

Furthermore, the contents of `examples/example-window-function-config.yaml` (unzipped from the 2.11 distro) does not match what's published in the documentation, causing some steps in the tutorial to not work as intended. 

![Screenshot 2023-04-10 at 4 38 47 PM](https://user-images.githubusercontent.com/9518298/231019324-a01df9da-732f-4197-9588-c63cd2a84e42.png)

### Minimal reproduce step

Follow the following guides in order using a fresh installation of pulsar 2.11:

1. https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-functions
2. https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-stateful-functions
3. https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-window-functions

### What did you expect to see?

The tutorial to work as described in the official documentation. 

### What did you see instead?

- Step No. 7 in https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-functions made no sense to me as a new user. 

- https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-stateful-functions can't be followed at all because the example file is missing. 

- I needed to update `examples/example-window-function-config.yaml` accordingly in order to run https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-window-functions. 

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
Thanks for reporting this issue @anguillanneuf!

Here we have two issues:

1. 2.11.0 distro doesn't have the file `examples/example-stateful-function-config.yaml`.

I can find it when building 3.0.0-SNAPSHOT distro at master. cc @Technoboy- (2.11.0 RM) do you have ideas why we miss this file in 2.11.0 distro? Also cc @poorbarcode (3.0.0 RM) as a reminder to check this file present when preparing 3.0.0 distro.

2. Document is inaccurate. Since you tick the box "I'm willing to submit a PR!", I'll double check if you're going to prepare a patch to apply all your suggestions above?

**Comment by anguillanneuf** [comment]
Thanks @tisonkun for distilling the docs issues into two separate problems.

>  if you're going to prepare a patch to apply all your suggestions above?

Below is what I can (know how) to do for 2.11 docs, are these sufficient though? 

- Step No. 7 in https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-functions made no sense to me as a new user.
  Remove this step completely. 

- https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-stateful-functions can't be followed at all because the example file is missing.
  Remove this section completely. 
  
- I needed to update examples/example-window-function-config.yaml accordingly in order to run https://pulsar.apache.org/docs/2.11.x/functions-quickstart/#start-window-functions.
  Edit docs to ask users to update the downloaded and unzipped `examples/example-window-function-config.yaml` according to https://github.com/apache/pulsar/blob/master/pulsar-functions/java-examples/src/main/resources/example-window-function-config.yaml

**Comment by Technoboy-** [comment]
As https://github.com/apache/pulsar/pull/18601 is not in branch-2.11, so the release of 2.11.x not contains this changes.
We need to improve the doc for 2.11.x

**Comment by tisonkun** [comment]
@Technoboy- Thanks for your investigation!

> We need to improve the doc for 2.11.x

After reviewing the changeset, I'd prefer to pick the changes of examples files and packaging logic in `bin.xml` to 2.11. Reverting 2.11 docs seems more complicated, and it benefits picking this non-functional improvement to 2.11.

**Comment by tisonkun** [comment]
@anguillanneuf Thanks for your update!

> Remove this step completely. No table service was started for me.

Sounds reasonable to be removed.

> examples/example-window-function-config.yaml

As @Technoboy- mentioned above, we will pick this change to the 2.11 branch, and thus the following 2.11 version should contain this file. So either we leave it as is or you can try to add a note for fetching the file on missing is viable to me.

**Comment by Technoboy-** [comment]
Thanks for reporting this, we will fix this in 2.11.1.
Currently, you can copy `example-stateful-function-config.yaml` from [here](https://github.com/apache/pulsar/blob/master/pulsar-functions/java-examples/src/main/resources/example-stateful-function-config.yaml) to examples folder

**Comment by Technoboy-** [comment]
Fixed by #20073

**Comment by tisonkun** [comment]
@anguillanneuf may you still send a patch for docs improvement?
